docs/api.md

```diff
@@ -0,0 +1,248 @@
+# API documentation
+
```

Comment:
Recommend having an introduction to the API in terms of what the user is trying to achieve. For example:

This API is for GOV.UK services that want to add user sign in / sign out and attribute storage functionality to their applications. When you have integrated this API with your service, you can sign in and authenticate your users' identities on GOV.UK.

Reply:
Added

> This API is for GOV.UK Publishing microservices (the things which make
> up www.gov.uk) to implement personalisation and user session
> management.  It is not for other government services.